analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/EnvironmentVariableScope.cs

```diff
@@ -34,6 +34,11 @@ public bool EnableConcurrentAnalysis
         {
             set => SetVariable(SonarDiagnosticAnalyzer.EnableConcurrentExecutionVariable, value.ToString());
         }
+
+        public bool EnableRazorAnalysis
```

Comment:
Since it is invoking a method on every "set", it cannot be a field. And if we change it to be a method, at the usage we would not be able to use it as nicely. We would need to create the object and invoke the method on it. Since it is only in the UTs + the environment variable will have at most 3 weeks to live before we remove it. I would not overly invest in this and wouldn't not change it.